src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -55,25 +45,50 @@ public static WorkloadResolver CreateForTests(IWorkloadManifestProvider manifest
             return new WorkloadResolver(manifestProvider, dotNetRootPaths, currentRuntimeIdentifiers);
         }
 
-        public WorkloadResolver CreateTempDirResolver(IWorkloadManifestProvider manifestProvider, string dotnetRootPath, string sdkVersion)
+        public WorkloadResolver CreateTempDirResolver(IWorkloadManifestProvider manifestProvider, string dotnetRootPath, string userHome, string sdkVersion)
+        {
+            string[] dotnetRootPaths = GetRootFolders(dotnetRootPath, userHome, sdkVersion);
+
+            return new WorkloadResolver(manifestProvider, dotnetRootPaths, _currentRuntimeIdentifiers);
+        }
+
+        private static string[] GetRootFolders(string dotnetRootPath, string userHome, string sdkFeatureBand)
         {
+            var userWorkloadFolder = GetUserWorkloadFolder(dotnetRootPath, sdkFeatureBand, userHome);
             var packRootEnvironmentVariable = Environment.GetEnvironmentVariable("DOTNETSDK_WORKLOAD_PACK_ROOTS");
+
             string[] dotnetRootPaths;
             if (!string.IsNullOrEmpty(packRootEnvironmentVariable))
             {
                 dotnetRootPaths = packRootEnvironmentVariable.Split(Path.PathSeparator).Append(dotnetRootPath).ToArray();
             }
+            else if (userWorkloadFolder != null && Directory.Exists(userWorkloadFolder))
+            {
+                dotnetRootPaths = new[] { dotnetRootPath, userWorkloadFolder };
```

Comment:
Likewies, if `DOTNETSDK_WORKLOAD_PACK_ROOTS` is set, I think we should still look for packs in the user workload folder if we would otherwise do so.